spec/facility_spec.rb

```diff
@@ -16,11 +24,125 @@
 
   describe '#add service' do
     it 'can add available services' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
       expect(@facility.services).to eq([])
       @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
+      @facility.add_service('Renew License')
+      @facility.add_service('Vehicle Registration')
+      @facility.add_service('Written Test')
+      @facility.add_service('Road Test')
+      expect(@facility.services).to eq(['New Drivers License', 'Renew License', 'Vehicle Registration', 'Written Test', 'Road Test'])
+    end
+  end
+
+  describe '#register vehicle' do
+    it 'starts with an empty array' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      expect(@facility.registered_vehicles).to eq([])
+    end
+
+    it 'can add vehicles to registered vehicles' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(@cruz)
+      @facility.register_vehicle(@bolt)
+      @facility.register_vehicle(@camaro)
+
+      expect(@facility.registered_vehicles).to eq([@cruz, @bolt, @camaro])
+    end
+
+    it 'sets todays date as registration date' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      @facility.register_vehicle(@cruz)
+
+      expect(@cruz.set_registration_date).to eq(Date.today)
+    end
+
+    it 'gives a license plate' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+      @facility.register_vehicle(@cruz)
+
+      expect(@cruz.give_plate).to eq(:regular)  
+    end
+  end
+
+
+  describe '#collect fees' do
+    it 'starts at 0' do
+    @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+     expect(@facility.collected_fees).to eq(0)
+    end
+
+    it 'increases with every vehicle registration' do
+      @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
       @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+      @facility.register_vehicle(@cruz)
+
+      expect(@facility.collected_fees).to eq(100)
```

Comment:
Yup, good test. Now we'd want to add one more vehicle to ensure that collected_fees will successfully add the additional registration fees.